submission/loglist_manager_test.go

```diff
@@ -62,7 +64,56 @@ func TestFirstRefresh(t *testing.T) {
 	case <-llm.LLUpdates:
 		return
 	case err := <-llm.Errors:
-		t.Errorf("llm.Run() returned error %q while expected none", err)
+		t.Errorf("llm.Run() emitted error %q while expected none", err)
+	case <-ctx.Done():
+		t.Errorf("llm.Run() on stub LogListRefresher expected to emit update, got none")
+	}
+}
+
+func TestSecondRefresh(t *testing.T) {
+	f, err := createTempFile(testdata.SampleLogList)
+	if err != nil {
+		t.Fatalf("createTempFile(%q) = (_, %q), want (_, nil)", testdata.SampleLogList, err)
+	}
+	defer os.Remove(f)
+
+	llr := NewLogListRefresher(f)
+	llm := NewLogListManager(llr)
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
+	defer cancel()
+	halfCtx, halfCancel := context.WithTimeout(context.Background(), time.Second)
+	defer halfCancel()
+
+	llm.Run(ctx, 300*time.Millisecond)
+	readFirst := false
+First:
+	for {
+		select {
+		case <-llm.LLUpdates:
+			if !readFirst {
+				readFirst = true
+			} else {
+				t.Errorf("llm.Run() emitted Log-list update when no updates happened")
+			}
+		case err := <-llm.Errors:
+			t.Errorf("llm.Run() remitted error %q while expected none", err)
+		case <-halfCtx.Done():
+			if !readFirst {
+				t.Errorf("llm.Run() emitted Log-list update when no updates happened")
```

Comment:
This message looks like a copy-paste error. In this case, if `readFirst == false`, it hasn't emitted any updates at all.

Reply:
Thanks for the catch. Updated the message.